app/grandchallenge/cases/image_builders/tiff.py

```diff
@@ -174,38 +208,40 @@ def _create_image_file(*, path: str, image: Image):
         )
 
 
-def _load_with_tiff(*, gc_file: GrandChallengeTiffFile):
+def _load_with_tiff(
+    *, gc_file: GrandChallengeTiffFile
+) -> GrandChallengeTiffFile:
     tiff_file = tifffile.TiffFile(str(gc_file.path.absolute()))
     gc_file = _extract_tags(gc_file=gc_file, pages=tiff_file.pages)
-    return tiff_file, gc_file
+    return gc_file
 
 
-def _load_with_open_slide(*, gc_file: GrandChallengeTiffFile, pk: UUID):
+def _load_and_create_dzi(
+    *, gc_file: GrandChallengeTiffFile
+) -> (str, GrandChallengeTiffFile):
     open_slide_file = openslide.open_slide(str(gc_file.path.absolute()))
     gc_file = _extract_openslide_properties(
         gc_file=gc_file, image=open_slide_file
     )
     gc_file.validate()
-    dzi_output = _create_dzi_images(gc_file=gc_file, pk=pk)
-    return dzi_output, gc_file
+    return _create_dzi_images(gc_file=gc_file)
 
 
 def _add_image_files(
-    *, tiff_file, gc_file, dzi_output, image, new_image_files
+    *, gc_file: GrandChallengeTiffFile, image: Image, new_image_files: List
 ):
-    if tiff_file:
-        new_image_files.append(
-            _create_image_file(path=str(gc_file.path.absolute()), image=image)
-        )
-
-    if dzi_output:
-        new_image_files.append(
-            _create_image_file(path=dzi_output + ".dzi", image=image)
-        )
+    new_image_files.append(
+        _create_image_file(path=str(gc_file.path.absolute()), image=image)
+    )
+    if gc_file.source_files:
+        for s in gc_file.source_files:
+            new_image_files.append(_create_image_file(path=s, image=image))
     return new_image_files
 
 
-def _add_folder_uploads(*, dzi_output, image, new_folder_upload):
+def _add_folder_uploads(
+    *, dzi_output: str, image: Image, new_folder_upload: List
```

Comment:
Aside: When you add a `List` annotation add the `List[<type>]`. No real benefit right now as we are not running mypy on the codebase, I'm unsure if we should even be adding type hints right now without static checking (https://mypy.readthedocs.io/en/stable/existing_code.html#start-small)

> If your codebase is large, pick a subset of your codebase (say, 5,000 to 50,000 lines) and run mypy only on this subset at first, _without any annotations_.

Emphasis theirs.